lib/deploy/functions/release.js

```diff
@@ -281,7 +282,7 @@ module.exports = function(context, options, payload) {
     _.chain(existingFunctions)
       .filter(function(functionInfo) {
         if (typeof functionInfo.labels === 'undefined') {
-          return functionInfo.sourceArchiveUrl === sourceUrl || functionInfo.sourceArchiveUrl === legacySourceUrl;
+          return functionInfo.sourceUploadUrl === sourceUrl || functionInfo.sourceArchiveUrl === stagingLegacySourceUrl || functionInfo.sourceArchiveUrl === legacySourceUrl;
```

Comment:
This is a really long line. Maybe it's worth breaking this up into temporaries that explain the tests you're making